src/common/telemetry/lib/Telemetry.cpp

```diff
@@ -0,0 +1,365 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "Telemetry.hpp"
+#include "ParameterSets.hpp"
+
+#include <algorithm>
+#include <chrono>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+#include <thread>
+
+using namespace MAT;
+
+LOGMANAGER_INSTANCE
+
+namespace Telemetry
+{
+
+OsConfigLogHandle TelemetryManager::m_log = NULL;
+MAT::ILogger* TelemetryManager::m_logger = nullptr;
+bool TelemetryManager::m_initialized = false;
+
+TelemetryManager::~TelemetryManager() noexcept
+{
+    Shutdown();
+}
+
+void TelemetryManager::SetupConfiguration(bool enableDebug, int teardownTime)
+{
+    auto& logConfig = LogManager::GetLogConfiguration();
+
+    logConfig[CFG_BOOL_ENABLE_TRACE] = enableDebug;
+    logConfig[CFG_INT_TRACE_LEVEL_MIN] = 0;
+    logConfig[CFG_INT_MAX_TEARDOWN_TIME] = teardownTime;
+
+    m_logger = LogManager::Initialize(API_KEY);
+}
+
+bool TelemetryManager::Initialize(bool enableDebug, int teardownTime)
+{
+    if (m_initialized)
+    {
+        return true;
+    }
```

Comment:
make this private and you probably want to init once in the constructor. 
avoid checks like this.